sklearn/svm/_base.py

```diff
@@ -638,10 +635,17 @@ def _validate_for_predict(self, X):
         # Fixes https://nvd.nist.gov/vuln/detail/CVE-2020-28975
         # Check that _n_support is consistent with support_vectors
         sv = self.support_vectors_
-        if not self._sparse and sv.size > 0 and self.n_support_.sum() != sv.shape[0]:
-            raise ValueError(
-                f"The internal representation of {self.__class__.__name__} was altered"
-            )
+        svm_type = LIBSVM_IMPL.index(self._impl)
+        if svm_type in (0, 1):
+            if (
+                not self._sparse
+                and sv.size > 0
+                and self.n_support_.sum() != sv.shape[0]
```

Comment:
This removes the error for SVR. I think we can remove the dependency on `self.n_support_`:

```python
        svm_type = LIBSVM_IMPL.index(self._impl)
        if svm_type in (0, 1):
            total_support = self._n_support.sum()
        else:
            total_support = self._n_support[0]

        if not self._sparse and sv.size > 0 and total_support != sv.shape[0]:
            raise ValueError(
                f"The internal representation of {self.__class__.__name__} was altered"
            )
```